NitroxClient/MonoBehaviours/Gui/Chat/PlayerChat.cs

```diff
@@ -0,0 +1,156 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using NitroxClient.GameLogic.ChatUI;
+using NitroxModel.Core;
+using UnityEngine;
+using UnityEngine.EventSystems;
+using UnityEngine.UI;
+
+namespace NitroxClient.MonoBehaviours.Gui.Chat
+{
+    public class PlayerChat : uGUI_InputGroup
+    {
+        private const int LINE_CHAR_LIMIT = 255;
+        private const int MESSAGE_LIMIT = 64;
+        private const float TOGGLED_TRANSPARENCY = 0.4f;
+        public const float CHAT_VISIBILITY_TIME_LENGTH = 10f;
+
+        private PlayerChatManager playerChatManager;
+        private CanvasGroup canvasGroup;
+        private HorizontalOrVerticalLayoutGroup[] layoutGroups;
+        private GameObject logEntryPrefab;
+        private Image[] backgroundImages;
+        private bool transparent;
+        private InputField inputField;
+
+        private Dictionary<ChatLogEntry, GameObject> entries = new Dictionary<ChatLogEntry, GameObject>();
+
+        public IEnumerator SetupChatComponents()
+        {
+            playerChatManager = NitroxServiceLocator.LocateService<PlayerChatManager>();
+
+            canvasGroup = GetComponent<CanvasGroup>();
+            canvasGroup.alpha = 0f;
+
+            layoutGroups = GetComponentsInChildren<HorizontalOrVerticalLayoutGroup>();
+
+            logEntryPrefab = GameObject.Find("ChatLogEntryPrefab");
+            logEntryPrefab.AddComponent<PlayerChatLogItem>();
+            logEntryPrefab.SetActive(false);
+
+            GetComponentsInChildren<Button>()[0].onClick.AddListener(ToggleTransparency);
+            GetComponentsInChildren<Button>()[1].gameObject.AddComponent<PlayerChatPinButton>();
+
+            inputField = GetComponentInChildren<InputField>();
+            inputField.gameObject.AddComponent<PlayerChatInputField>();
+            inputField.GetComponentInChildren<Button>().onClick.AddListener(playerChatManager.SendMessage);
+
+            backgroundImages = new[]
+            {
+                transform.GetChild(0).GetComponent<Image>(), transform.GetChild(1).GetComponent<Image>(), transform.GetChild(3).GetComponent<Image>()
+            };
+            yield return new WaitForEndOfFrame();
+        }
+
+        public void WriteLogEntry(ChatLogEntry chatLogEntry)
+        {
+            chatLogEntry.MessageText = SanitizeMessage(chatLogEntry.MessageText);
+
+            if (entries.Count == MESSAGE_LIMIT)
+            {
+                Destroy(entries.Values.First());
+                entries.Remove(entries.Keys.First());
+            }
+
+            GameObject newChatLogEntry = Instantiate(logEntryPrefab, logEntryPrefab.transform.parent);
```

Comment:
New entry should not keep their world pos.
```suggestion
            GameObject newChatLogEntry = Instantiate(logEntryPrefab, logEntryPrefab.transform.parent, false);
```